x-pack/plugins/canvas/common/lib/autocomplete.d.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  AvailableFunctions,
+  Function,
+  ArgumentType,
+} from '../../canvas_plugin_src/functions/types';
+
+export interface AutocompleteFunctionSuggestion {
+  type: 'function';
+  text: string;
+  start: number;
+  end: number;
+  fnDef: AvailableFunctions;
+}
+
+export interface AutocompleteArgNameSuggestion {
+  type: 'arg';
+  text: string;
+  start: number;
+  end: number;
+  argDef: ArgumentType<any>;
```

Comment:
The `any` generic here is probably not the right way to go about this. 

Should we have an AvailableArguments type like we have AvailableFunctions?